[Domaintools] Initial Iris Release v2.0.0

wesleya opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
N/A

## Description
This is an initial PR for version 2.0 of the DomainTools Iris pack. This release includes:

* New commands for Iris Enrich, Whois History, Hosting History, Reverse Whois, Domain Profile, Whois, Reverse IP, Reverse Name Server. 
* New Scripts for Extract and Enrich, and Extract and Investigate.
* New Playbooks for checking domains by High Risk Score, Tags, and Search Hash
* Bug fixes and improvements

We have more changes incoming for this 2.0 release, but wanted to get some initial feedback on what we have so far.

## Must have
- [x] Tests
- [x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=29474) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=29474) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=29474) it.</sub>

wesleya commented:
We are using a new Docker image in this PR. The PR for the new image is at https://github.com/demisto/dockerfiles/pull/19710

But I wasn't sure what to use as the file for the `dockerimage` value in this PR while the new image is still under review. I've been using a local image for testing. Please let me know if I need to make any changes here before you can test/review. Thank you!

MichaelYochpaz commented:
@wesleya I'll get the other PR for the new Docker image approved & merged first, and then you can just update it here in this PR before approval (if we don't do that, it should fail tests and validations for using a non-existing Docker image).

Could you please also sign the [CLA mentioned here](https://github.com/demisto/content/pull/29474#issuecomment-1706984438)? We can start the review process before it's signed.

wesleya commented:
@MichaelYochpaz I've updated our app to use the `demisto/vendors-sdk` image which includes our sdk. And I believe I've signed the CLA. We will reach out to you to schedule a meet to go over the changes. Thank you!

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsSetIndicatorTable/set_indicator_table_data.py:
```diff
@@ -0,0 +1,86 @@
+        except Exception:
+            reputation = "Unknown"
+
+        demisto_indicator = {
+            "type": "Domain",
+            "value": domain_name,
+            "source": "DomainTools",
+            "reputation": reputation,
+            "seenNow": "true",
+            "ipaddresses": json.dumps(domaintools_hosting_data.get("IPAddresses")),
+            "ipcountrycode": domaintools_hosting_data.get("IPCountryCode"),
+            "mailservers": json.dumps(domaintools_hosting_data.get("MailServers")),
+            "spfrecord": domaintools_hosting_data.get("SPFRecord"),
+            "nameservers": domaintools_hosting_data.get("NameServers"),
+            "sslcertificate": json.dumps(
+                domaintools_hosting_data.get("SSLCertificate")
+            ),
+            "soaemail": domaintools_identity_data.get("SOAEmail"),
+            "sslcertificateemail": domaintools_identity_data.get("SSLCertificateEmail"),
+            "emaildomains": domaintools_identity_data.get("EmailDomains"),
+            "additionalwhoisemails": domaintools_identity_data.get(
+                "AdditionalWhoisEmails"
+            ),
+            "domainage": domain_age,
+        }
+        demisto.executeCommand("createNewIndicator", demisto_indicator)
```
Use the `Common.Domain` object from `CommonServerPython` for creating an indicator.

briluza replied:
Just wanted to clarify here. Did you mean for us to continue using the demisto.executeCommand("createNewIndicator", indicator) call, but pass in a Common.Domain object instead of a python dict? Or did you mean for us to use a different method for creating the new indicator? We didn't see a way to save the Common.Domain indicator once it was created.

Here's a snippet we are using 
` return CommandResults(readable_output=human_readable_str, indicator=domain_indicator)`

checking it has an output however, it doesn't create an indicator.

MichaelYochpaz replied:
This is probably because you're passing to the `indicator` parameter a dict instead of an indicator object (in this case, I believe it should be a `Common.Domain` object?).

You can see an example for a proper `CommandResults` object with an indicator [here](https://github.com/demisto/content/blob/master/Packs/CommonScripts/Scripts/ResolveShortenedURL/ResolveShortenedURL.py#L356-L369).

briluza replied:
sorry for the confusion, we are passing a `Common.Domain` object to command result. 
here's the `domain_indicator` value.
 `domain_indicator = Common.Domain(domain=domain_name,
                                         creation_date=create_date,
                                         dbot_score=dbot_score
                                         )`

MichaelYochpaz replied:
I'm not sure indicators should appear in the debug panel there.
They should be added to XSOAR's "Threat Intel" panel.

Passing a custom dict like that is non-conventional and can cause all-sort of unexpected bugs & issues, since there is some additional processing done on indicator objects on the `CommonServerPython` module (when running `return_Results` with a `CommandResults` object).

briluza replied:
Thanks @MichaelYochpaz, is it good idea if we just create our own/custom indicator type to cater the fields we are defining when creating an indicator? 

we are adding this custom fields:
<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-d83d8736-7fff-7645-ebaf-abaea1e8bde9"><div dir="ltr" style="margin-left:36pt;" align="left">

additionalWhoisEmails
--
domainAge
emailDomains
ipAddresses
mailServers
nameServers
soaEmail
spfRecord
sslCertificate

</div></b>

briluza replied:
As per discussed internally, we will move forward on creating a custom indicator type e.g. (Domaintools Iris) and add the custom fields.

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsExtractAndEnrich/extract_and_enrich.py:
Like I mentioned in the demo, it might be a good idea to just add this functionality inside the `domaintoolsiris-enrich` command, instead of a separate script.

Then instead of running `ExtractDomainFromUrlAndEmail` I believe we have a `CommonServerPython` function for extracting the domain from URLs / emails you can use for the value passed in this parameter. @Ni-Knight might be able to confirm.

This is up for you to decide, though, feel free to resolve this comment if you prefer to keep the script.

MichaelYochpaz replied:
We automatically detect indicators like URL and E-mails (and also IPs, hashes, etc), and enrich them using reputation commands (like the `!domain` command), so this script is a bit redundant, considering the indicator will be enriched anyway once the indicator is extracted by the server.

Ni-Knight replied:
Just clarifying here, there is no reason not to build this wrapper but it is built wrong. Instead of `ExtractDomainFromUrlAndEmail` you should be using `extractIndicators` which is the server side extraction script and the one that is constantly updated.

Instead of `domaintoolsiris-enrich` just call `!domain` and add a flag telling it to use `Iris` (`!domain domain=domaintools.com using=DomainToolsIris`)

wesleya replied:
@Ni-Knight we tested with `extractIndicators` but noticed it would leave in subdomains which doesn't always have results in our API. Is it okay to continue using `ExtractDomainFromUrlAndEmail` here? 

we have one script for extracting a domain and enriching with `domain` and another for extracting a domain and enriching with `domaintoolsiris-enrich`. Is it okay to keep both for slightly different use cases?

MichaelYochpaz reviewed (commented):
Hey, thank you for the contribution :)

I left quite a few comments, but most of them are about the same issues / concerns.
I may have missed a few additional sections with these issues (about `CommandResults`, creating indicators, and using `.get()`).
If you notice any other similar cases where I haven't noticed, please fix those as well.

Also, make sure to add the missing release notes (can be generated using `demisto-sdk update-release-notes`)

You're more than welcome to reach out to me in the DFIR Slack / E-mail if you encounter any issues / have any questions or need help.

MichaelYochpaz commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -89,36 +98,14 @@ def get_dbot_score(proximity_score, age, threat_profile_score):
     else:
         return 1
 
+def should_include_context(include_context):
+    if include_context is False:
+        return False
```
We've discussed this and concluded that this is problematic, since reputation commands run automatically by the server to enrich indicators, and adding this flag will make it behave unexpectedly.

Please update it to return indicators in any case.
As for the performance issues you had, which are why you added this, please reach out to me on Slack, and we'll see what the issue is and how we can resolve it.

wesleya replied:
We've updated the `should_include_context` to default to `True`. We thought this would allow automatic runs to work as expected, while giving users the option to turn it off for a single manual run if they run into performance issues. What do you think?

wesleya commented:
@MichaelYochpaz Thank you for the feedback! We have this work scheduled in our upcoming sprint.

melamedbn commented:
Hi @wesleya,

Firstly, thank you for your contribution on the playbooks. After reviewing, I've a few suggestions to enhance the content and user experience:

### Playbook: DomainTools Check Domain Risk Score By Iris Tags

1. **Text Files in Playbook Folder**: The two text files, `DT_HighRiskScore_Threshold.txt` and `Iris_Tags_To_Query.txt`, seem out of place. If they are meant to act as lists, please place them under a 'Lists' folder. However, for enhanced user flexibility, it might be more beneficial to use hardcoded values, playbook inputs, or avoid lists.
2. **Playbook Inputs**: Task 1 could benefit from playbook inputs for better flexibility.
3. **Playbook Flow with High Risk Score**: Consider adding an alert or conditional task. This way, users have the option to review the domain and not halt all playbook executions.
![image](https://github.com/demisto/content/assets/28757135/cb159a00-1459-460b-807b-6b8492cd6014)

4. **Typo**: Found a typo in the playbook description: "ivestigate" should be "investigate".
5. **Outputs**: The playbook currently has no outputs. Is this intentional?

### Playbook: DomainTools Check New Domains by Iris Hash

1. **Description Clarity**: The `search_hash` description could be clearer for better understanding.
2. **Typo**: Noticed an extra period in the playbook description.
3. **Complex Context**: When possible, opt for complex context (as indicated in the attached screenshot). This is relevant for tasks 1 and 5.
![image](https://github.com/demisto/content/assets/28757135/5bf84e0c-9561-4936-9e22-39649c1f8908)

4. **Playbook Flow with High Risk Score**: Similar to the first playbook, consider adding an alert or conditional task for high-risk score domains.
5. **Unused Arguments**: There are arguments which aren't being utilized. Perhaps integrating them into the playbook inputs might provide users with more flexibility.

I truly appreciate your hard work and dedication. These suggestions are meant to help refine and improve the playbooks for optimal user experience. Looking forward to seeing the updates!

Warm regards,

Ben

MichaelYochpaz commented:
Hey @wesleya, any updates?
Please let me know if you encountered any issues and need help.

wesleya commented:
Hi @MichaelYochpaz apologies for the late reply we were pulled away for a bit and returning to this now. Thank you again for your feedback! I think most of it makes sense so far and we are currently working through your suggestions. We did run into a few questions we wanted to ask you about:

> Then instead of running ExtractDomainFromUrlAndEmail I believe we have a CommonServerPython function for extracting the domain from URLs / emails you can use for the value passed in this parameter.

We weren't able to find a CommonServerPython function for extracting domains in the reference docs, can you point us in the right direction here? Otherwise we are also happy to leave this as a separate script.

> We automatically detect indicators like URL and E-mails (and also IPs, hashes, etc), and enrich them using reputation commands (like the !domain command), so this script is a bit redundant, considering the indicator will be enriched anyway once the indicator is extracted by the server.

I'm not sure what we're missing here but we don't seem to be getting any automated enrichment for domains from the  `!ExtractDomainFromUrlAndEmail` script. see linked video:
https://www.loom.com/share/8b38d7e12d6044cd9de79aed2617d912?sid=8d412bbd-e104-4881-b86b-51e33605c8fa

Is there a setting we need to configure to get that working? Would it still be okay to include our own `DomainToolsExtractAndEnrich` script instead of relying on the context generated by the automated enrich? Since we format the human readable result with some useful info and links not available in the context.

> Use the Common.Domain object from CommonServerPython for creating an indicator.

Just wanted to clarify here. Did you mean for us to continue using the `demisto.executeCommand("createNewIndicator", indicator)` call, but pass in a `Common.Domain` object instead of a python dict? Or did you mean for us to use a different method for creating the new indicator? We didn't see a way to save the `Common.Domain` indicator once it was created.

> We've discussed this and concluded that this is problematic, since reputation commands run automatically by the server to enrich indicators, and adding this flag will make it behave unexpectedly.
Please update it to return indicators in any case.
As for the performance issues you had, which are why you added this, please reach out to me on Slack, and we'll see what the issue is and how we can resolve it.

Here is a video of the performance issue we ran into with our `!domain` command:
https://www.loom.com/share/fa260189c5bf488daaef8af7be2a0b99?sid=e3e15507-2367-458d-82d6-cf529436c69c

We are open to your guidance is here, just let us know what options you prefer. thank you!

Ni-Knight commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -441,23 +430,381 @@ def domain_profile_command():
+    if not should_include_context(include_context):
+        all_context = None
+
+    results = CommandResults(
+        outputs_prefix='',
+        outputs_key_field='Name',
+        outputs=all_context,
+        readable_output=all_human_readable_output,
+        ignore_auto_extract=True,
+        raw_response=all_raw_response
+    )
```
How is `domain_enrich_command` different form `!domain using=Iris`?

wesleya replied:
@MichaelYochpaz Updated! We are putting all data that fits in the Common.Domain object through the indicator parameter. And we have a separate context that has custom data fields we pass through the outputs parameter

@Ni-Knight the `domain_enrich_command` uses a different api endpoint that has slightly different data and rate limits than the one used in `!domain`. enrich is more of a bulk enrich use case, investigate is more of a human scale use case.

Ni-Knight commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -584,46 +929,279 @@ def domain_pivot_command():
 
 
+def whois_history_command():
+    domain = demisto.args().get('domain')
```
As domain here is mandatory It should be changed

```suggestion
    domain = demisto.args()['domain']
```

wesleya replied:
updated

Ni-Knight commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -584,46 +929,279 @@ def domain_pivot_command():
+    return str
 
 
+def whois_history_command():
```
Not sure if its mandatory but I would strongly recommend adding function docstrings for future reference and easier debugging. I would also suggest including in those an example to how the data is outputted.

wesleya replied:
updated

MichaelYochpaz commented:
@wesleya Please write questions about existing comments on their respective comment conversations. It's difficult to find which code section / comment you're referring to in that format.

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -25,11 +27,13 @@
 YOUNG_DOMAIN_TIMEFRAME = int(demisto.params().get('young_domain_timeframe'))
 VERIFY_CERT = not demisto.params().get('insecure', False)
 PROXIES = handle_proxy()
+GUIDED_PIVOT_THRESHOLD = int(demisto.params().get('pivot_threshold'))
```
updated

wesleya commented:
@MichaelYochpaz @Ni-Knight we've taken a pass at addressing your feedback. Let us know if we're missing anything, thank you!

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsCheckTags/check_tags.py:
```diff
@@ -0,0 +1,41 @@
+    domain_tags_set = set([domain_tag['label'] for domain_tag in domain_tags])
+
+    malicious_tags = args['malicious_tags']
+    malicious_tags_set = set(json.loads(malicious_tags))
```
Please set a const variable at the top of this file instead of an external file.
Something like:
```python
MALICIOUS_TAGS = ["xyz", "footyphishing","Demistodemo","test_tag", "blocklist"]
```

briluza replied:
this is supposed to be a dynamic vars which users can freely change. Will be moving this as an argument rather getting the values from an external file. Thank you!

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsCheckTags/check_tags.py:
```diff
@@ -0,0 +1,41 @@
+    except Exception as ex:
+        demisto.error(traceback.format_exc())
+        return_error(
+            f"Failed to execute check_pivotable_domains. Error: {str(ex)}")
```
```suggestion
            f"Failed to execute check_tags. Error: {str(ex)}")
```

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsCheckLastEnrichment/check_last_enrichment.py:
```diff
@@ -0,0 +1,37 @@
+    except Exception as ex:
+        demisto.error(traceback.format_exc())
+        return_error(
+            f"Failed to execute set_indicator_table_data. Error: {str(ex)}")
```
```suggestion
            f"Failed to execute CheckLastEnrichment. Error: {str(ex)}")
```

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsCheckLastEnrichment/check_last_enrichment.yml:
```diff
@@ -0,0 +1,28 @@
+outputs:
+- contextPath: 'yes'
+  description: Refresh Enrichment Data
+  type: Unknown
```
```suggestion
  type: String
```

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsCheckLastEnrichment/check_last_enrichment.yml:
```diff
@@ -0,0 +1,28 @@
+  type: Unknown
+- contextPath: 'no'
+  description: Don't Refresh Enrichment Data
+  type: Unknown
```
```suggestion
  type: String
```

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsAddDomainRiskScoreToContext/add_domain_riskscore_to_context.py:
```diff
@@ -0,0 +1,41 @@
+        return_results(add_domain_riskscore_to_context(demisto.args()))
+    except Exception as ex:
+        demisto.error(traceback.format_exc())  # print the traceback
+        return_error(f'Failed to execute BaseScript. Error: {str(ex)}')
```
```suggestion
        return_error(f'Failed to execute AddDomainRiskScoreContext. Error: {str(ex)}')
```

melamedbn commented:
> Hi @wesleya,
> 
> Firstly, thank you for your contribution on the playbooks. After reviewing, I've a few suggestions to enhance the content and user experience:
> 
> ### Playbook: DomainTools Check Domain Risk Score By Iris Tags
> 1. **Text Files in Playbook Folder**: The two text files, `DT_HighRiskScore_Threshold.txt` and `Iris_Tags_To_Query.txt`, seem out of place. If they are meant to act as lists, please place them under a 'Lists' folder. However, for enhanced user flexibility, it might be more beneficial to use hardcoded values, playbook inputs, or avoid lists.
> 2. **Playbook Inputs**: Task 1 could benefit from playbook inputs for better flexibility.
> 3. **Playbook Flow with High Risk Score**: Consider adding an alert or conditional task. This way, users have the option to review the domain and not halt all playbook executions.
>    ![image](https://user-images.githubusercontent.com/28757135/269231943-cb159a00-1459-460b-807b-6b8492cd6014.png)
> 4. **Typo**: Found a typo in the playbook description: "ivestigate" should be "investigate".
> 5. **Outputs**: The playbook currently has no outputs. Is this intentional?
> 
> ### Playbook: DomainTools Check New Domains by Iris Hash
> 1. **Description Clarity**: The `search_hash` description could be clearer for better understanding.
> 2. **Typo**: Noticed an extra period in the playbook description.
> 3. **Complex Context**: When possible, opt for complex context (as indicated in the attached screenshot). This is relevant for tasks 1 and 5.
>    ![image](https://user-images.githubusercontent.com/28757135/269232330-5bf84e0c-9561-4936-9e22-39649c1f8908.png)
> 4. **Playbook Flow with High Risk Score**: Similar to the first playbook, consider adding an alert or conditional task for high-risk score domains.
> 5. **Unused Arguments**: There are arguments which aren't being utilized. Perhaps integrating them into the playbook inputs might provide users with more flexibility.
> 
> I truly appreciate your hard work and dedication. These suggestions are meant to help refine and improve the playbooks for optimal user experience. Looking forward to seeing the updates!
> 
> Warm regards,
> 
> Ben

@wesleya Hi, did you get a chance to go over my review notes?

wesleya commented:
Hi @MichaelYochpaz @Ni-Knight @melamedbn, we've updated the PR with changes to address your feedback. Please let us know if you have any additional feedback.

MichaelYochpaz commented:
@wesleya  @briluza Please note that there are many unaddressed notes.
Perhaps they're collapsed by GitHub ("X hidden conversations") and you missed them?

A few examples:
https://github.com/demisto/content/pull/29474#discussion_r1320770123
https://github.com/demisto/content/pull/29474#discussion_r1320772281
https://github.com/demisto/content/pull/29474#discussion_r1320772332
https://github.com/demisto/content/pull/29474#discussion_r1320772536
https://github.com/demisto/content/pull/29474#discussion_r1324351467

(there are more)

wesleya commented:
@MichaelYochpaz apologies for the overlook there. I'll pick these up and reply when it's updated. Thank you!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -441,23 +430,381 @@ def domain_profile_command():
+        readable_output=all_human_readable_output,
+        ignore_auto_extract=True,
+        raw_response=all_raw_response
+    )
```
Updated, thank you for the guidance here. We are putting all data that fits in the `Common.Domain` object through the `indicator` parameter. And we have a separate context that has custom data fields we pass through the `outputs` parameter

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -584,46 +929,279 @@ def domain_pivot_command():
+        readable_output=human_readable,
+        ignore_auto_extract=True
+    )
+    return_results(results)
```
Our mistake here, we're not returning a domain indicator. We're returning a list of changes to whois of a domain over time. Updated the naming and prefix to reflect that

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -584,46 +929,279 @@ def domain_pivot_command():
+def domain_profile_command():
+    domain = demisto.args().get('domain')
+    results = domain_profile(domain)
+    return_results({'response': results})
```
We removed this command as the `CommandResults` was so similar to the `!domain` result.

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -584,46 +929,279 @@ def domain_pivot_command():
+        outputs=context,
+        readable_output=human_readable,
+        ignore_auto_extract=True
+    )
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -144,6 +131,49 @@ def prune_context_data(data_obj):
+def format_dns_grid(type, dns_dict):
+    return [{
+    "type": type,
+    "ip": ','.join([ip.get('value') for ip in item.get('ip')]),
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -144,6 +131,49 @@ def prune_context_data(data_obj):
+    return result
+
+def format_tags(tags):
+    return ' '.join([tag.get('label') for tag in tags])
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -255,27 +290,45 @@ def create_domain_context_outputs(domain_result):
         'DomainStatus': domain_status,
         'ExpirationDate': expiration_date,
-        'NameServers': [name_server.get('host', {}).get('value') for name_server in name_servers],
+        'DNS':  [{"type": 'DNS', "ip": ip.get('address').get('value')} for ip in ip_addresses]
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -255,27 +290,45 @@ def create_domain_context_outputs(domain_result):
+            'Organization': registrant_org,
+        },
+        'Geo': {
+            'Country': ' '.join([ip.get('country_code').get('value') for ip in ip_addresses])
```
updated!

wesleya commented:
@MichaelYochpaz I believe I've addressed all of the missed notes (you're right this PR has grown larger than expected). Please let me know if I missed anything.

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsAddDomainRiskScoreToContext/add_domain_riskscore_to_context.py:
```diff
@@ -0,0 +1,42 @@
+    result = {"Name": domain_name, "OverallRiskScore": domain_risk_score}
+
+    return CommandResults(
+        outputs_prefix='HighRiskPivotedDomains',
```
Having a different root key than the name of the script can get confusing to users (who might see that key in context and be unsure which step of the playbook added it).
I think this will be a good way to make it more clear.

```suggestion
        outputs_prefix='AddDomainRiskScoreToContext.HighRiskPivotedDomains',
```

wesleya replied:
updated!

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsAddDomainRiskScoreToContext/add_domain_riskscore_to_context.yml:
```diff
@@ -0,0 +1,18 @@
+- name: domaintools_data
+  required: true
+outputs:
+- contextPath: HighRiskPivotedDomains.Name
+  description: The domain name
+- contextPath: HighRiskPivotedDomains.OverallRiskScore
```
If changing the `output_prefix` as I suggested, make sure to update the `contextPath` here as well.

wesleya replied:
updated!

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsCheckPivotableDomains/check_pivotable_domains.yml:
```diff
@@ -0,0 +1,99 @@
+args:
+- name: domaintools_data
+  required: true
+- name: max_registrant_contact_name_count
+  defaultValue: "200"
+- name: max_registrant_org_count
+  defaultValue: "200"
+- name: max_registrar_count
+  defaultValue: "200"
+- name: max_ssl_info_organization_count
+  defaultValue: "200"
+- name: max_ssl_info_hash_count
+  defaultValue: "350"
+- name: max_ssl_email_count
+  defaultValue: "350"
+- name: max_ssl_subject_count
+  defaultValue: "350"
+- name: max_name_server_host_count
+  defaultValue: "250"
+- name: max_name_server_ip_count
+  defaultValue: "250"
+- name: max_name_server_domain_count
+  defaultValue: "250"
+- name: max_soa_email_count
+  defaultValue: "200"
+- name: max_ip_address_count
+  defaultValue: "200"
+- name: max_mx_ip_count
+  defaultValue: "200"
+- name: max_mx_host_count
+  defaultValue: "200"
+- name: max_mx_domain_count
+  defaultValue: "200"
+- name: max_google_adsense_count
+  defaultValue: "200"
+- name: max_google_analytics_count
+  defaultValue: "200"
```
Please add descriptions to these arguments.

wesleya replied:
updated!

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsCheckTags/Domaintools_Malicious_Tags.json:
Please remove this file

wesleya replied:
removed!

MichaelYochpaz commented on Packs/DomainTools_Iris/Scripts/DomainToolsExtractAndEnrich/extract_and_enrich.yml:
```diff
@@ -0,0 +1,333 @@
+  secret: false
+comment: Resolves a URL or fully qualified domain name (FQDN) and looks up a complete profile of the domain on the DomainTools Iris Enrich API.
+commonfields:
+  id: domainExtractAndEnrich
+  version: -1
+enabled: true
+name: domainExtractAndEnrich
```
Please follow the Pascal Case convention
```suggestion
  id: DomainExtractAndEnrich
  version: -1
enabled: true
name: DomainExtractAndEnrich
```

wesleya replied:
updated!

MichaelYochpaz reviewed (commented):
Hey, I added some additional notes.

Please also:
* Move the following directories:
`Packs/DomainTools_Iris/Integrations/DomainTools_Iris/IndicatorFields`
`Packs/DomainTools_Iris/Integrations/DomainTools_Iris/IndicatorTypes`
`Packs/DomainTools_Iris/Integrations/DomainTools_Iris/Layouts`
To be under `Packs/DomainTools_Iris` (same level as `Integrations`, not under it).

* Rename the script files to match their names (for example, `add_domain_riskscore_to_context.py` to `AddDomainRiskScoreContext.py`. Both `py` and `yml` files should be updated).

* Merge from upstream master (or sync your fork and then merge master) so that our CI steps will run properly, and we'll be able to see if the PR passes all validations.

* Address the following review note:
https://github.com/demisto/content/pull/29474/files#r1320767182

wesleya commented:
@MichaelYochpaz thank you for the quick review. I'll pick these up and reply again when it's updated. Thank you!

briluza commented on Packs/DomainTools_Iris/Scripts/DomainToolsSetIndicatorTable/set_indicator_table_data.py:
```diff
@@ -0,0 +1,102 @@
+
+        human_readable_str = "Data for {} enriched.".format(domain_name)
+
+    return CommandResults(readable_output=human_readable_str)
```
Hi @MichaelYochpaz  just an fyi. this script is outdated and we use a custom indicator type named `DomainTools Iris` .

MichaelYochpaz commented:
@wesleya Following the merge from master, the validation checks now ran successfully.

Please note the failing validations and fix them too:
https://app.circleci.com/pipelines/github/demisto/content/314204/workflows/d5344e8a-92db-434c-95fb-337c1a2c89a1/jobs/727294

If the detected secrets are not actual secrets, you can add them to a `.secrets-ignore` file. See `Packs/Rapid7_Nexpose/.secrets-ignore` for an example for such file, and [the docs](https://xsoar.pan.dev/docs/packs/packs-format#secrets-ignore).
If you need any help with fixing that or any of the other validations, please let me know.

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -225,13 +261,13 @@ def create_domain_context_outputs(domain_result):
             'WebsiteResponseCode': website_response,
             'GoogleAdsenseTrackingCode': google_adsense,
             'GoogleAnalyticTrackingCode': google_analytics,
-            'AlexaRank': alexa_rank,
```
Hi @MichaelYochpaz, I'm still working on this one. Do you know why we might be running into this error:

`[ERROR] Packs/DomainTools_Iris/IndicatorFields/indicatorfields-DomainTools_Iris.json: [BA102] - File DomainTools_Iris/IndicatorFields/indicatorfields-DomainTools_Iris.json is not supported in the validate command.
The validate command supports: Integrations, Scripts, Playbooks, Incident fields, Incident types, Indicator fields, Indicator types, Objects fields, Object types, Object modules, Images, Release notes, Layouts, Jobs, Wizards, Descriptions And Modeling Rules.`

Here is the command I'm trying to run:
`demisto-sdk update-release-notes -i Packs/DomainTools_Iris/Integrations/DomainTools_Iris/ -v 2.0.0 -bc`

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -89,36 +99,8 @@ def get_dbot_score(proximity_score, age, threat_profile_score):
-        return None
-
+def should_include_context(include_context):
+    return include_context == "true"
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -574,54 +1003,358 @@ def threat_profile_command():
+        )
+    )
+
+    return_results(command_results_list)
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -441,38 +480,407 @@ def domain_profile_command():
+            )
+
+
+    return_results(command_results_list)
```
updated! I combined the indicator and domaintools context into a single `CommandResult`, but kept the `command_results_list` list because we support multiple domains here. Is this okay?

wesleya commented:
Hi @MichaelYochpaz, I believe we've addressed all comments from your review plus moving and renaming the indicators, layouts, and scripts.

We will take a look at the validation errors from the CI run and submit another update to get those fixed as well.

We're also still looking into an error we run into during the release notes update. Please let us know if we're missing something obvious: https://github.com/demisto/content/pull/29474/files#r1320767182

Thank you!

MichaelYochpaz commented on Packs/DomainTools_Iris/IndicatorFields/indicatorfields-DomainTools_Iris.json:
File should be named `indicatorfield-DomainTools_Iris.json`

MichaelYochpaz commented on Packs/DomainTools_Iris/IndicatorTypes/reputations-Domaintools_Iris.json:
File should be named `reputation-Domaintools_Iris.json`

MichaelYochpaz reviewed (commented):
I believe the wrong file names are the cause for the issue here
https://github.com/demisto/content/pull/29474/files#r1320767182

wesleya commented:
Hi @MichaelYochpa, we're making our way through the validation errors and adding unit tests. Thank you for your help there.  I had another question about some of the errors we're getting in our `DomainTools_Iris.yml` file:

1. We're getting a bunch of errors about backwards compatibility, even with the `breakingChanges` field set to `True` in our release notes - https://app.circleci.com/pipelines/github/demisto/content/316600/workflows/b5f170d8-9304-44c0-80a9-22f37840c902/jobs/728916?invite=true#step-119-5284_112

2. A new required field - https://app.circleci.com/pipelines/github/demisto/content/316584/workflows/d57898f6-ab75-4bcc-b938-46547c04c308/jobs/728880?invite=true#step-119-6299_177

Thank you again for your guidance!

MichaelYochpaz commented:
@wesleya These validations will remain there, they won't disappear when using BC changelogs.
You can ignore them for now, but do ensure that you don't break BC for anything you didn't intend to, and that everything is documented in the release notes.

wesleya commented:
@MichaelYochpaz wanted to give you a heads up that we had to update this PR with some additional features in order to address some bugs we ran into by trying to keep new features in a separate release: https://github.com/DomainTools/content/pull/4. 

We are continuing to address CI errors, we appreciate your patience with this. Please let us know if you have any questions.  thank you!

wesleya commented:
Hi @MichaelYochpaz, we’re in a good spot to check in. I believe we’ve addressed all of the validation issues except for a few we wanted to ask about:

1. The `Secrets` check is giving us this error: Running secrets validation against origin/master
```
Running secrets validation against origin/master
Packs/DomainTools_Iris/Classifiers/classifier-DomainTools_Iris_-_Classifier.json has a structure issue of file type .json
Expected object or value
```
2. We’re getting a few mypy errors related to `arg_to_number` possible returning None. I “think” these might be false positives because we always pass in a default value to `arg_to_number` but wanted to get your thoughts. For example:
```
[2023-11-15 17:49:20,619] [INFO] CheckPivotableDomains
/home/circleci/project/Packs/DomainTools_Iris/Scripts/CheckPivotableDomains/CheckPivotableDomains.py:344:9: error:
Argument 3 to "check_pivotable_ssl_info" has incompatible type "Optional[int]";
expected "int"  [arg-type]
            max_ssl_info_organization_count,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 1 error in 1 file (checked 1 source file)
```
3. Is there any other feedback we need to address in order to approve this PR?

Again, appreciate your patience with this as we're learning the Cortex platform for the first time. Please let us know if you have any questions. thank you!

MichaelYochpaz commented:
@wesleya

1 - I don't see this error for the last commit, seems like there's a different one I'm not sure about. Pull from master and see if it changes. If it doesn't, I'll look further into it.

3 - Not at the moment, but since there were many changes to the code since the last review, we'll probably have to re-review it once you finish with the changes and everything is ready (from your side, and the build is passing without any validation erros / issues).

wesleya commented:
@MichaelYochpaz just checking in to let you know we're working on fixing all validation errors and will let you know when we're finished with changes and everything is ready. Thank you!

wesleya commented:
@MichaelYochpaz  we're done with the changes and everything is ready for re-review. Looks like all validation checks are passing other than the breaking changes checks we discussed earlier: https://github.com/demisto/content/pull/29474#issuecomment-1807091284

Let us know if there's anything else you need at this time, thank you!

MichaelYochpaz commented:
Hey @wesleya, we're currently re-reviewing the PR following the changes :)
Do note though that It will probably take longer than usual, due to how unusually big the PR is.

wesleya commented:
Here is a demo of the changes we submitted in this PR: https://info.domaintools.com/Instant-Infrastructure-Intel_VideoPage.html

The demo starts about 20 minutes into the video. We're also available for a live demo, but thought this might help in the mean time.

Ni-Knight commented on Packs/DomainTools_Iris/IndicatorFields/indicatorfield-additionalwhoisemails.json:
```diff
@@ -0,0 +1,31 @@
+    "id": "indicator_additionalwhoisemails",
+    "version": -1,
+    "modified": "2023-10-16T17:13:01.885961592Z",
+    "name": "Additional Whois Emails",
```
Please add a prefix here.

cc - @MichaelYochpaz

wesleya replied:
updated!

Ni-Knight commented on Packs/DomainTools_Iris/IndicatorFields/indicatorfield-domainage.json:
```diff
@@ -0,0 +1,31 @@
+    "id": "indicator_domainage",
+    "version": -1,
+    "modified": "2023-10-16T17:13:01.885961592Z",
+    "name": "Domain Age",
```
Prefix needed

wesleya replied:
@Ni-Knight apologies I'm not sure what you mean for this one. Do you have a link to docs, or a good example I can follow here?

wesleya replied:
I updated the id, name, cliName fields and filename with a prefix of `domaintoolsiris`. Let me know if you had something else in mind here.

melamedbn commented:
Hi @wesleya,

I've been reviewing the DomainTools Iris content you've submitted and I'd like to share some feedback to ensure we maximize its effectiveness and user-friendliness.

### Playbooks:

#### DomainTools Check New Domains by Iris Hash
- **Input Clarification**: It would be beneficial to add an example to the `search_hash` input field. This addition can help users better understand the expected format and usage, leading to a smoother experience.

#### DomainTools Associate Indicator to Incident
- **Purpose and Functionality**: Could you provide more insight into the necessity and operational mechanism of this playbook? A brief explanation within the playbook description, detailing its intended use and benefits, would greatly aid users in understanding its goal.

#### Indicator fields/type:
Please address @Ni-Knight feedback.

Thanks for your ongoing efforts in refining the content. If you have any questions or need further clarification on the feedback, please feel free to reach out.

Best regards,
Ben

wesleya commented on Packs/DomainTools_Iris/Layouts/layoutscontainer-DomainTools_Iris_Indicator_Layout.json:
makes sense, updated to match domain layout as best we can.

wesleya commented on Packs/DomainTools_Iris/ReleaseNotes/2_0_0.md:
release notes updated with the new command names

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -303,338 +392,1216 @@ def domain_profile(domain):
+        indicator=domain_indicator,
+        readable_output=human_readable,
+        ignore_auto_extract=True
+    )
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -303,338 +392,1216 @@ def domain_profile(domain):
+def to_camel_case(value):
+    str = f' {value.strip()}'
+    str = re.sub(r' ([a-z,A-Z])', lambda g: g.group(1).upper(), str)
+    return str
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -303,338 +392,1216 @@ def domain_profile(domain):
+        http_request('iris-investigate', {'domains': 'demisto.com'})
         demisto.results('ok')
     except Exception:
         raise
```
updated!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.yml:
```diff
@@ -1009,286 +1539,533 @@ script:
+      name: query
+      required: true
+    description: The DomainTools Parsed Whois API provides parsed information extracted from the raw Whois record. The API is optimized to quickly retrieve the Whois record, group important data together and return a well-structured format. The Parsed Whois API is ideal for anyone wishing to search for, index, or cross-reference data from one or multiple Whois records.
+    name: whois
```
renamed!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.yml:
```diff
@@ -1009,286 +1539,533 @@ script:
+      - "true"
+      - "false"
+    description: The DomainTools Reverse Whois API provides a list of domain names that share the same Registrant Information. You can enter terms that describe a domain owner, like an email address or a company name, and you’ll get a list of domain names that have your search terms listed in the Whois record.
+    name: reverseWhois
```
renamed!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.yml:
```diff
@@ -1009,286 +1539,533 @@ script:
+    deprecated: false
+    description: Hosting History will list IP address, name server and registrar history.
+    execution: false
+    name: hostingHistory
```
renamed!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.yml:
```diff
@@ -1009,286 +1539,533 @@ script:
+    deprecated: false
+    description: The DomainTools Whois History API endpoint returns up to 100 historical Whois records associated with a domain name.
+    execution: false
+    name: whoisHistory
```
renamed!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -303,338 +392,1216 @@ def domain_profile(domain):
+        'First Seen',
+        'Server Type',
+        'Popularity'
+    ]
```
yes that makes more sense, updated to constant!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
```diff
@@ -303,338 +392,1216 @@ def domain_profile(domain):
+        iris_tags=iris_tags_params
+    )
+
+    return True
```
remmoved!

wesleya commented on Packs/DomainTools_Iris/Integrations/DomainTools_Iris/DomainTools_Iris.py:
we ran the `demisto-sdk pre-commit` command and fixed all of the type hinting errors for now. We plan to fix the remaining errors in a follow up release, but let us know if any of these fixes are required.

wesleya commented:
Hi @melamedbn @MichaelYochpaz  @Ni-Knight , thank you taking the time to review. I believe I've either merged your suggested commit directly, or manually addressed all of the comments in this round. Please let me know if you need any other changes. Thank you!

MichaelYochpaz commented:
@wesleya Looks good from my side, good job :)
Waiting for @melamedbn's and @Ni-Knight's greenlight to continue to a demo, if there are no additional notes.

melamedbn commented:
> Hi @wesleya,
> 
> I've been reviewing the DomainTools Iris content you've submitted and I'd like to share some feedback to ensure we maximize its effectiveness and user-friendliness.
> 
> ### Playbooks:
> #### DomainTools Check New Domains by Iris Hash
> * **Input Clarification**: It would be beneficial to add an example to the `search_hash` input field. This addition can help users better understand the expected format and usage, leading to a smoother experience.
> 
> #### DomainTools Associate Indicator to Incident
> * **Purpose and Functionality**: Could you provide more insight into the necessity and operational mechanism of this playbook? A brief explanation within the playbook description, detailing its intended use and benefits, would greatly aid users in understanding its goal.
> 
> #### Indicator fields/type:
> Please address @Ni-Knight feedback.
> 
> Thanks for your ongoing efforts in refining the content. If you have any questions or need further clarification on the feedback, please feel free to reach out.
> 
> Best regards, Ben

Hi @wesleya,

I didn't find any fix/comment for the feedback I've shared. Waiting for your response and we'll be able to move on with the demo.

Thanks,
Ben

wesleya commented:
@melamedbn apologies for the missing comment! I've made the changes below to address your comments:

**DomainTools Check New Domains by Iris Hash** - I added an example hash to the description of the `search_hash` input field - https://github.com/demisto/content/pull/29474/files#diff-a4686259fbc14da00c5ba1d544ea5cd73ef527b3a2701b06bb51dd665ad0aadfR269

**DomainTools Associate Indicator to Incident** - I updated the description with more details - https://github.com/demisto/content/pull/29474/files#diff-266c429cce3006dcbb22720c8c8fba4fb690707c4941806516599abc14f613faR4

Let me know if you had something else in mind for either of these comments, thank you!

melamedbn commented:
> @melamedbn apologies for the missing comment! I've made the changes below to address your comments:
> 
> **DomainTools Check New Domains by Iris Hash** - I added an example hash to the description of the `search_hash` input field - https://github.com/demisto/content/pull/29474/files#diff-a4686259fbc14da00c5ba1d544ea5cd73ef527b3a2701b06bb51dd665ad0aadfR269
> 
> **DomainTools Associate Indicator to Incident** - I updated the description with more details - https://github.com/demisto/content/pull/29474/files#diff-266c429cce3006dcbb22720c8c8fba4fb690707c4941806516599abc14f613faR4
> 
> Let me know if you had something else in mind for either of these comments, thank you!

Hi @wesleya,

Thanks! 
I'm not sure about the associate indicators playbook and script, we already have an automation which link an indicator to an incident. Can you explain the benefit/advantages of both the playbook and script in the PR?

wesleya commented:
@melamedbn We have monitoring playbooks that add new indicators to context on a configured schedule. For example new domains that match a risk score or age threshold set by the user. Our `AssociateIndicatorsToIncident` script uses the cortex command under the hood to associate the indicators created by our monitors with a user configured incident. It might be easier to show you in the demo you mentioned earlier? And then you can let us know if there's a better way to accomplish our use case?

We are available this week Thursday or Friday between 8:00 AM - 10:00 AM Pacific. Please suggest a different time if you prefer, we're pretty flexible. Thank you!

melamedbn commented:
> @melamedbn We have monitoring playbooks that add new indicators to context on a configured schedule. For example new domains that match a risk score or age threshold set by the user. Our `AssociateIndicatorsToIncident` script uses the cortex command under the hood to associate the indicators created by our monitors with a user configured incident. It might be easier to show you in the demo you mentioned earlier? And then you can let us know if there's a better way to accomplish our use case?
> 
> We are available this week Thursday or Friday between 8:00 AM - 10:00 AM Pacific. Please suggest a different time if you prefer, we're pretty flexible. Thank you!

Seeing it in a demo sounds good. as for the meeting time, unfortunately I won't be available in both Thursday and Friday this week.

wesleya commented:
> Hey @wesleya, good job on the demo :)
> Following what we said about the !domain command, we've checked and seems like overriding the default domain command and changing its description & parameters to something unconventional will just cause confusion, as these reputation commands are not built for such usages.
> 
> For example, if a user has multiple integrations configured that support the domain reputation command, the description of the command that is shown to the user will be randomly selected (probably according to order of enablement or something similar), and sometimes it might show the specific description you wrote with the limitations, and suggest the include_context parameter, and sometimes the default one we use for other commands. For any integration, as long as there are multiple domain commands with different attributes.
> 
> So basically, what we want is to line up the domain reputation command with the others - same description and no additional parameters. To handle limitations, you can just raise an error from within the code, something like "Enrichment of over 1,000 domains is not supported", and also write that down on the README file of the integration under the 'domain' command.
> 
> As for the include_context parameter - if you think it's important to have such an option, create a different command (a domaintools specific one with the prefix) that does the same as the domain reputation command, but there you can add whatever description and parameters you'd like, and we don't need to stay in line with a convention like the reputation commands.

@MichaelYochpaz thank you for the demo review. This all makes sense, we'll work on these changes as well as the playbook changes we talked about during the demo. Thank you!

MichaelYochpaz commented:
Hey @wesleya,
Last Thursday, December 20th, there was an update in the content repo where some old commits were modified.
This caused this PR (and all the others) to have a lot of additional file changes.

To resolve the issue, please do the following:
1. Backup all of your changes locally (very important! In case something goes wrong).
2. Enter your forked repository page in GitHub.
3. At the top of the page, you should see something like the following screenshot. Please ensure you are on the master branch.
![1](https://github.com/demisto/content/assets/8832013/4da4eade-e3a8-4e2f-9665-282e79c27df2)

4. Click on the “Sync fork” button on the top right side of the page.
5. The following message will appear ("This branch has conflicts that must be resolved").
![2](https://github.com/demisto/content/assets/8832013/8712862d-48ff-44f8-9d99-f12c475a433e)

6. Click on the “Discard commits” button. Note that for this action to work, you must have permission to do a force push in the forked repo. If you don’t have one, ask the forked repo owner to do that.

7. Select your working branch and repeat the same steps (4-6).
8. Check your PR, and assure all of your changes are there. If they are - you finished the update process.
9. If your PR does not show your changes, open a new PR with the changes that you backed-up at the start.
11. If a new PR is opened, link the current one there in the description, and tag me there.

Sorry for the hassle, and thank you for your contribution.

wesleya commented:
@MichaelYochpaz thanks for the heads up. Steps 1-6 worked great. But I don't see the sync button for step 7:

> Select your working branch and repeat the same steps (4-6).

This is the on the branch we have the open PR for. Here is what I see on our branch (DomainTools:domaintools-iris-release-v2.0.0) right now:
![Screenshot 2023-12-26 at 2 37 24 PM](https://github.com/demisto/content/assets/18468747/70468c72-f628-43a8-b2bd-9b1f61e05f1b)

My initial thought was that I just need to merge master into the domaintools-iris-release-v2.0.0 branch, but thought I'd check with you first.

Let me know if I misunderstood, and what step I should take next. Or if this is unexpected we can open up a new PR. Apologies for the slow response as our team is on holiday this week. Thank you!

MichaelYochpaz commented:
@wesleya This issue is due to changes that were made to the git history of the content repo, so I suspect merging won't resolve this issue, and that a rebase will be required. But that also might not be flawless and might require some tweaking.
If you think you'll be able to resolve the issue yourself by rebasing onto the updated `master` branch, you can try that, and then if you fail to do that, just open a new PR.

Just make sure to properly backup your branch's changes beforehand, and assure the PR has the exact same code you had before.

No pressure / anything urgent, you can do that after the holiday. Just wanted to give you an update so that you'll understand what happened to the PR.

wesleya commented:
@MichaelYochpaz it did turn out to be easier to re-create our PR after running into some trouble with the rebase. I will comment at you when it is ready for your review. Thank you!

wesleya commented:
@MichaelYochpaz yes you may close this PR since we will use https://github.com/demisto/content/pull/31952 moving forward. It's still under our internal testing, I'll tag you when it's ready for your review. Thank you!
